python/src/aiconfig/schema.py

```diff
@@ -819,6 +819,14 @@ def _update_model_name_for_prompt(self, model_name: str, prompt_name: str):
             model_settings: InferenceSettings = (
                 prompt.metadata.model.settings or {}
             )
+            # TODO (rossdanlm): This is a hack so that when Hugging Face task
+            # (defined as model_name) is switched, the "model" field in setting
+            # is also updated and set to None so that it resets to default model
+            # for that HF task. We're setting the entire settings just to empty
+            # (using defaults from input schema) to make UI client updating easier
+            if hasattr(model_settings, "model"):
+                # model_settings["model"] = None
```

Comment:
Can you clarify why it's easier to clear the whole thing instead of just setting the model to None? It's just a few lines of code difference but I think clearing is going to be a worse UX in some case (we used to clear and then updated to not clear based on product feedback)

Reply:
I just wanted to get a solution out quickly and ensure I could quickly test. Pretty tired so haven't been thinking too deeply